JumpscaleCore/tools/threebot/ThreebotExplorer.py

```diff
@@ -161,3 +167,98 @@ def sign(nacl, *args):
         self._log_info("registration of threebot '{%s}' done" % name)
 
         return record1
+
+    def _reservation_create(self, flist="", hub_url="", environment={}, entrypoint=""):
+        """
+        Create a reservation object with container workload from schema url of "tfgrid.reservation.1".
+        The parameters passed will be added to the container workload
+
+        :param flist: link to flist to create a container from
+        :type flist: str
+        :param hub_url: link to hub
+        :type hub_url: str
+        :param environment: environment variables to be passed to the contiainer to be created eg : {"USER":"test_user"}
+        :type environment: dict
+        :param entrypoint: entrypoint to container start
+        :type entrypoint: str
+        :return: reservation object
+        :return type: obj
+        """
+        reservation_model = j.data.schema.get_from_url(url="tfgrid.reservation.1")
+
+        reservation = reservation_model.new()
+        reservation.customer_tid = j.tools.threebot.me.default.tid
+        reservation.data_reservation.expiration_provisioning = int(j.data.time.epoch + 30 * 60)
+        reservation.data_reservation.expiration_reservation = int(j.data.time.epoch + 50 * 60)
+
+        # Create container workload
+        container = reservation.data_reservation.containers.new()
+        container.node_id = "1"
```

Comment:
this also need to be configurable and there is no default value possible.

Reply:
node_id is now configurable and should be given